### ISSUE TITLE
Long distance block catchup failed

### ISSUE THREAD

**Issue by Geod24** [bosagora/agora#2455]
```
2021-09-07 00:54:54,665 Error [agora.consensus.Ledger] - Validator 0x210f6551d648a4da654da116b100e941e434e4f232b8579439c2ef64b04819bd2782eb3524c7a29c38c347cdf26006bccac54a58a58f103ae7eb5b252eb53b64 (index: 0, address: boa1xrval7gwhjz4k9raqukcnv2n4rl4fxt74m2y9eay6l5mqdf4gntnzhhscrh) missing pre-image for height 1 (known: 0)
2021-09-07 00:54:54,665 Error [agora.consensus.Ledger] - Validator 0x3b44d65edb3361dd91441ab4f449eeda55644026624c4b8ae12ecf0264fa8a228dbf672ef97e2c4f87fb98ad7099e17b7f9ba7dbe8479672066912b1ea24ba77 (index: 1, address: boa1xzval2a3cdxv28n6slr62wlczslk3juvk7cu05qt3z55ty2rlfqfc6egsh2) missing pre-image for height 1 (known: 0)
2021-09-07 00:54:54,665 Error [agora.consensus.Ledger] - Validator 0x7bacc99e9bf827f0fa6dc6a77303d2e6ba6f1591277b896a9305a9e200853986fe9527fd551077a4ac2b511633ada4190a7b82cddaf606171336e1efba87ea8f (index: 2, address: boa1xzval3ah8z7ewhuzx6mywveyr79f24w49rdypwgurhjkr8z2ke2mycftv9n) missing pre-image for height 1 (known: 0)
2021-09-07 00:54:54,666 Error [agora.consensus.Ledger] - Validator 0x9b2726e79f05abc107b6531486a46c977414e13ed9f3ee994ec14504964f86fcf9464055b891b9c34020feb72535c300ff19e8b5167eb9d202db1a053d746b2c (index: 3, address: boa1xrval6hd8szdektyz69fnqjwqfejhu4rvrpwlahh9rhaazzpvs5g6lh34l5) missing pre-image for height 1 (known: 0)
2021-09-07 00:54:54,666 Error [agora.consensus.Ledger] - Validator 0xab19131ad8974a20881e2cd0798684a06ca0054160735cdf67fe8ee5a0eb4e28e9bf3f4c735f9ed3da958778978c86b409b8d133f30992141f0ac7e01e7f1255 (index: 4, address: boa1xrval5rzmma29zh4aqgv3mvcarhwa0w8rgthy3l9vaj3fywf9894ycmjkm8) missing pre-image for height 1 (known: 0)
2021-09-07 00:54:54,666 Error [agora.consensus.Ledger] - Validator 0xdb7664caba94c8d4602c10992c13176307e1e05361c150217166ee77fc4af9bf176f31dc61aba61e634dfc0b4c5f729d59e604607f61c9f66b10c6841f972a0a (index: 5, address: boa1xzval4nvru2ej9m0rptq7hatukkavemryvct4f8smyy3ky9ct5u0s8w6gfy) missing pre-image for height 1 (known: 0)
2021-09-07 00:54:54,674 Error [agora.consensus.Ledger] - validateBlock: Exception thrown by getRandomSeed while externalizing valid block: agora.common.Ensure.FormattedException@source/agora/consensus/Ledger.d(1011): Requested pre-images for validators that are not yet known
----------------
??:? @safe void agora.common.Ensure.ensure!().ensure(bool, immutable(char)[], immutable(char)[], int) [0x108432734]
??:? @safe geod24.bitblob.BitBlob!(64uL).BitBlob agora.consensus.Ledger.Ledger.getRandomSeed(in agora.common.Types.Height, in uint[]) [0x1084c0cfb]
??:? nothrow @safe immutable(char)[] agora.consensus.Ledger.Ledger.validateBlock(in ref agora.consensus.data.Block.Block) [0x1084bff73]
??:? @safe immutable(char)[] agora.consensus.Ledger.Ledger.acceptBlock(in ref agora.consensus.data.Block.Block) [0x1087bd009]
??:? @safe agora.common.Types.Height agora.node.FullNode.FullNode.addBlocks(const(agora.consensus.data.Block.Block)[], const(agora.consensus.data.PreImageInfo.PreImageInfo)[]) [0x1087bcb14]
??:? nothrow void agora.network.Manager.NetworkManager.getBlocksFrom(agora.common.Types.Height, agora.common.Types.Height delegate(const(agora.consensus.data.Block.Block)[], const(agora.consensus.data.PreImageInfo.PreImageInfo)[]) @safe) [0x1083a1d0f]
??:? nothrow void agora.node.FullNode.FullNode.catchupTask() [0x1087b5f4a]
??:? nothrow @trusted void vibe.core.core.setTimer(core.time.Duration, void delegate(), bool).__lambda4() [0x108ffe3d3]
??:? nothrow @safe void vibe.core.core.createTimer(void delegate() nothrow @safe).C.opCall(vibe.core.core.Timer).__lambda2(vibe.core.core.Timer) [0x108ffe5a1]
??:? nothrow void vibe.core.task.TaskFuncInfo.set!(void delegate(vibe.core.core.Timer) nothrow @safe, vibe.core.core.Timer).set(ref void delegate(vibe.core.core.Timer) nothrow @safe, ref vibe.core.core.Timer).callDelegate(ref vibe.core.task.TaskFuncInfo) [0x109046e55]
??:? void vibe.core.task.TaskFuncInfo.call() [0x10904474b]
??:? nothrow void vibe.core.task.TaskFiber.run() [0x109043e68]
??:? fiber_entryPoint [0x1092b37a3]
```

```
./build/agora -c devel/config-fullnode.yaml
```

**Comment by Geod24** [comment]
Most likely fixed by #2519 . Will close once the test system reaches 1k block and we can still sync.

**Comment by Geod24** [comment]
Confirmed.
